lib/bundler/settings.rb

```diff
@@ -201,14 +201,21 @@ def set_key(key, value, hash, file)
         hash.delete(key) if value.nil?
         SharedHelpers.filesystem_access(file) do |p|
           FileUtils.mkdir_p(p.dirname)
-          require "bundler/psyched_yaml"
-          File.open(p, "w") {|f| f.puts YAML.dump(hash) }
+          p.open("w") {|f| f.write(serialize_hash(hash)) }
         end
       end
 
       value
     end
 
+    def serialize_hash(hash)
+      yaml = String.new("---\n")
+      hash.each do |key, value|
+        yaml << key << ": " << value.to_s.gsub(/\s+/, " ").inspect << "\n"
```

Comment:
There aren't any current cases where the `.bundle/config` would have or require something more complex than just simple key --> value mapping is there?